en/Plugins/Guides/Manage plugin lifecycle.md

```diff
@@ -0,0 +1,293 @@
+---
+permalink: plugins/guides/lifecycle-management
+---
+
+When developing a plugin, you will often create resources such as event listeners, DOM elements, intervals or workers. While setting these up is often straightforward, it is equally important to properly clean them up when your plugin is unloaded. Neglecting this can lead to memory leaks, dangling event handlers, or unexpected behavior.
+
+
+## What you'll learn
+
+After completing this guide, you'll be able to:
+
+- Manage the lifecycle of resources in your plugin.
+- Use the `Component` class to simplify cleanup and organization.
+- Avoid common pitfalls that lead to resource leaks or stale data.
+- Understand how components and subcomponents interact in Obsidian.
+
+
+## Overview
+
+Imagine your plugin needs to respond to window resize events.
+A simple (but incorrect implementation!) might look like this:
+
+```ts
+// Bad
+class MyPlugin extends Plugin {
+	onload() {
+		window.addEventListener("resize", onResize);
+	}
+}
+```
+
+This works, but it introduces a problem: when the user disabled your plugin, the resize listener remains active! Only when Obsidian is restarted will it be removed. While this might be a fairly innocuous example, in more complex scenarios, this can lead to errors and memory leaks.
+
+A more responsible solution would be to manually remove the listener during `onunload`:
+
+```ts
+// Better
+class MyPlugin extends Plugin {
+	onload() {
+		window.addEventListener("resize", onResize);
+	}
+
+	onunload() {
+		window.removeEventListener("resize", onResize);
+	}
+}
+```
+
+While this works, it quickly becomes repetitive and error-prone. Fortunately, Obsidian's `Component` system offers a better solution.
```

Comment:
This seems a little awkward since the example with onload and onunload is _already_ using functions of Component.